### DISCUSSION TITLE
Migrating Comments from Disqus and Utterances to Giscus | /en/2022/12/disqus-to-giscus/

### DISCUSSION THREAD

**Discussion by giscus** [rbind/yihui#1457] [Comments]
Like Rob, I have wished to ditch Disqus for a long time, which I have used for more than a decade. I was aware of Maëlle Salmon&rsquo;s work in 2019 (and Nan Xiao&rsquo;s in 2020), but hesitated at &hellip;

https://yihui.org/en/2022/12/disqus-to-giscus/

**Comment by asimumba** [comment]
Great to come back here after a few months in the wild.

**Reply by yihui** [reply]
I'll definitely blog more often this year. Watch out for the flood! :)

**Comment by szcf-weiya** [comment]
Yihui, 最近我也用上了 giscus，刚刚才发现我那边 discussion (https://github.com/szcf-weiya/blog-comment/discussions) 默认只显示 pathname，而你的既有 title 和 pathname，

> modify the discussion titles (to make them clearer instead of only having a pathname in them).

检查了 giscus 的配置，跟你的一致；

然后试图在你仓库里面找文中提到的 `gh::gh_gql` 的 scripts，但没有找到；

也跑到 blogdown 那边看了，也没有找到（虽然我用的不是 blogdown

很好奇你是怎么实现的？只是在本地跑处理代码吗？

另：新年快乐！期待中文博客~

**Reply by yihui** [reply]
代码不用找，我没有发出来，真的是太乱了，我自己都不想看第二遍，而且我是一边跑一边改的，运行的顺序也是乱七八糟，根本不知道哪几句先运行、哪几句后运行的了。修改标题用的就是这一节提到的 mutation：https://yihui.org/en/2022/12/disqus-to-giscus/#deal-with-character-escaping-in-github-graphql 但这个批量修改仅限于已有的评论。对新评论，Giscus 无法自动创建我想要的标题（即：title + pathname），我两年前提了需求甚至赞助了作者，但至今没有实现：https://github.com/giscus/giscus/issues/832 所以我这里的新评论我都是手动修改标题的。

**Reply by szcf-weiya** [reply]
明白了，谢谢！！
